File: 3183-find-the-k-or-of-an-array/find-the-k-or-of-an-array.py
class Solution:
    def findKOr(self, nums: List[int], k: int) -> int:
        max_bits = max(nums).bit_length()

        result = 0
        
        for bit in range(max_bits):
            count = 0
            for num in nums:
                if num & (1 << bit):
                    count += 1
            if count >= k:
                result |= (1 << bit)
        return result

File: 3183-find-the-k-or-of-an-array/find-the-k-or-of-an-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findKOr = function(nums, k) {
    let result = 0;

    for (let bit = 0; bit < 31; bit++) {
        let count = 0;
        for (const num of nums) {
            if ((num >> bit) & 1) {
                count += 1;
            }
        }
        if (count >= k) {
            result |= (1 << bit);

        }
    }
    return result;
};